output/influxoutput.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (C) 2023 Seaburr
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+//go:build darwin
+
+package output
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/fvbommel/sortorder"
+)
+
+type InfluxOutput struct {
+	writer io.Writer
+}
+
+func NewInfluxOutput() Output {
+	o := InfluxOutput{}
+	o.writer = io.Writer(os.Stdout)
+
+	return o
+}
+
+func (io InfluxOutput) All() {
+	all := GetAll()
+
+	keys := make([]string, 0, len(all))
+	for k := range all {
+		keys = append(keys, k)
+	}
+	sort.Sort(sortorder.Natural(keys))
+
+	for _, key := range keys {
+		value := all[key]
+		if smcdata, ok := value.(map[string]interface{}); ok {
+			io.print(key, smcdata)
+		}
+	}
+}
+
+func (io InfluxOutput) Battery() {
+	io.print("Battery", GetBattery())
+}
+
+func (io InfluxOutput) Current() {
+	io.print("Current", GetCurrent())
+}
+
+func (io InfluxOutput) Fans() {
+	io.print("Fans", GetFans())
+}
+
+func (io InfluxOutput) Power() {
+	io.print("Power", GetPower())
+}
+
+func (io InfluxOutput) Temperature() {
+	io.print("Temperature", GetTemperature())
+}
+
+func (io InfluxOutput) Voltage() {
+	io.print("Voltage", GetVoltage())
+}
+
+func influx_string_convert(s string) string {
+	s = strings.Replace(s, " ", "_", -1)
+	s = strings.ToLower(s)
+	return s
+}
+
+func influx_strip_unit(s string) string {
+	s = strings.Split(s, " ")[0]
+	return s
+}
+
+func (io InfluxOutput) print(name string, smcdata map[string]interface{}) {
+	if len(smcdata) != 0 {
+		ct := time.Now().UnixNano()
+
+		keys := make([]string, 0, len(smcdata))
+		for k := range smcdata {
+			keys = append(keys, k)
+		}
+
+		for _, k := range keys {
+			v := smcdata[k]
+			if value, ok := v.(map[string]interface{}); ok {
+				key := influx_string_convert(fmt.Sprintf("key=%s", value["key"]))
+				value := fmt.Sprintf("value=%v", value["value"])
+				if key == "key=" {
+					fmt.Println(
+						fmt.Sprintf("%v,sensortype=%s %s %d",
```

Comment:
I wouldn't do `fmt.Println()` with `fmt.Sprintf()` as an argument, you can simplify that into one `fmt.Printf()` with a newline. Also it would make sense to switch on `value["key"]` early, as allocating a new string into `key` and switching on `key` looks a bit too late?

Reply:
Thank you for the review and feedback! I'll do some work to clean this up and update my branch next week.

Reply:
I've made a number of changes to account for your comments. I'm also including the unit as a tag on the data now instead of throwing it away. Let me know what you think. I will start looking into add unit tests next.